community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledEquivalenceUtils.java

```diff
@@ -262,6 +266,29 @@ public static int hashCode( Object[] array )
         }
     }
 
+    /**
+     * Calculate hash code of a AnyValue[] value
+     * @param array the value to compute hash code for
+     * @return the hash code of the given value
+     */
+    public static int hashCode( AnyValue[] array )
+    {
```

Comment:
Can't we just use

```
Arrays.hashCode( array );
```

I don't like this trick.

Reply:
Only for this method or do you mean change all of the hashCode methods that use this?